node-tests/acceptance/build-test.js

```diff
@@ -1,31 +1,349 @@
+      let app = new AddonTestApp();
+      let engineName = 'eager';
+      let addonName = 'nested';
+
+      yield app.create('engine-testing', { noFixtures: true });
+      let engine = yield InRepoEngine.generate(app, engineName, { lazy: false });
+      let addon = yield engine.generateNestedAddon(addonName);
+
+      addon.writeFixture({
+        app: {
+          'foo.js': '// should wind up in engine namespace'
+        },
+        addon: {
+          components: {
+            'bar.js': '// should wind up in addon namespace'
+          },
+          templates: {
+            components: {
+              'bar.hbs': '<h1>bar</h1>'
+            }
+          }
+        }
+      });
+
+      let output = yield build(app);
+
+      // Verify app files in addon wind up in engine's namespace in vendor.js
+      output.contains('assets/vendor.js', moduleMatcher(`${engineName}/foo`));
+
+      // Verify addon files in addon wind up in addon's namespace in vendor.js
+      output.contains('assets/vendor.js', moduleMatcher(`${addonName}/components/bar`));
+      output.contains('assets/vendor.js', moduleMatcher(`${addonName}/templates/components/bar`));
+    }));
+
+    it('correctly builds addon in lazy engine', co.wrap(function* () {
+      let app = new AddonTestApp();
+      let engineName = 'lazy';
+      let addonName = 'nested';
+
+      yield app.create('engine-testing', { noFixtures: true });
+      let engine = yield InRepoEngine.generate(app, engineName, { lazy: true });
+      let addon = yield engine.generateNestedAddon(addonName);
+
+      addon.writeFixture({
+        app: {
+          'foo.js': '// should wind up in engine namespace'
+        },
+        addon: {
+          components: {
+            'bar.js': '// should wind up in addon namespace'
+          },
+          templates: {
+            components: {
+              'bar.hbs': '<h1>bar</h1>'
+            }
+          }
+        }
+      });
+
+      let output = yield build(app);
+
+      // Verify app files in addon wind up in engine's namespace in engine.js
+      output.contains(`engines-dist/${engineName}/assets/engine.js`, moduleMatcher(`${engineName}/foo`));
+
+      // Verify addon files in addon wind up in addon's namespace in enigne-vendor.js
+      output.contains(`engines-dist/${engineName}/assets/engine-vendor.js`, moduleMatcher(`${addonName}/components/bar`));
+      output.contains(`engines-dist/${engineName}/assets/engine-vendor.js`, moduleMatcher(`${addonName}/templates/components/bar`));
+    }));
+
+    it('invokes trees and executes hooks in the proper order and handles imports for engines', co.wrap(function* () {
+      let app = new AddonTestApp();
+      let engineName = 'tree-invocation-order';
+
+      yield app.create('engine-testing', { noFixtures: true });
+      let engine = yield InRepoEngine.generate(app, engineName, { lazy: true });
+
+      let fixture = fixturify.readSync(path.resolve(__dirname, '../fixtures/tree-invocation-order'));
```

Comment:
I'd prefer to store this in a single file, instead of using `fixturify.readSync`.  I find that it makes it much much easier to grok if I can see the structure in one place.

The file can look like:

```js
const stripIndent = require('common-tags').stripIndent;

module.exports = {
  'some/path': stripIndent`
    whatever text contents go here
  `
}
```

Then just pass that in to `engine.writeFixture`